docs/user_guides/llm/vertexai/README.md

```diff
@@ -0,0 +1,102 @@
+# Using LLMs hosted on Vertex AI
+
+This guide shows an example that queries LLMs hosted on Vertex AI. This guide includes an example GuardRails configuration that is a variation of the ABC bot defined in the the [Getting Started Guide](../../getting_started). The only change is that this changes the model being used `gemini-1.0-pro` using the `vertexai` engine.
+
+In order to use Vertex AI, you need to perform some initial setup with the Google Cloud Platform (GCP).
+1. Create a GCP account: The following [page](https://cloud.google.com/docs/get-started) provides more information about the Google Cloud Platform and how to get started. In your account [create a project](https://cloud.google.com/resource-manager/docs/creating-managing-projects) and [set up billing for it](https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_an_existing_project)
+2. Install the `gcloud` CLI ([guide](https://cloud.google.com/sdk/docs/install)). Note that although 3.8 - 3.12 are listed as supported, [this error](https://stackoverflow.com/questions/77316716/gcloud-modulenotfounderror-no-module-named-imp) occurs on Python 3.12. This guide was tested using Python 3.10.2.
+3. Create a service account following [this guide](https://cloud.google.com/iam/docs/service-accounts-create) and grant it the role of `Vertex AI Service Agent`.
+4. Create and download a service account key for the service account ([guide](https://cloud.google.com/iam/docs/keys-create-delete)).
+5. Enable the Vertex AI API ([guide](https://cloud.google.com/vertex-ai/docs/start/cloud-environment#:~:text=Enable%20Vertex%20AI%20APIs,-In%20the%20Google&text=Click%20Enable%20All%20Recommended%20APIs,the%20APIs%20are%20being%20enabled.))
+6. Install additional python libraries needed to call Vertex AI using `pip install google-cloud-aiplatform>=1.38.0`
```

Comment:
We should move the steps about enabling and testing(the snippet below with `start_chat()`) VertexAI into a separate guide in `docs/user_guides/advanced`. This guide should mention: "This guide assumes you have configured VertexAI. If not, refer to this guide.". It should only mention installing the dependencies and setting the environment variable.

Reply:
Updated in commit [5319dbc5b89201e1e4d7046386183dbd7c3b4a12](https://github.com/NVIDIA/NeMo-Guardrails/pull/402/commits/5319dbc5b89201e1e4d7046386183dbd7c3b4a12).